libflux/src/flux/semantic/flatbuffers/tests.rs

```diff
@@ -0,0 +1,952 @@
+extern crate flatbuffers;
+extern crate walkdir;
+
+use super::semantic_generated::fbsemantic;
+use crate::ast;
+use crate::semantic;
+use crate::semantic::analyze;
+use chrono::FixedOffset;
+
+#[test]
+fn test_serialize() {
+    let f = vec![
+        crate::parser::parse_string(
+            "test1",
+            r#"
+package testpkg
+import other "my_other_pkg"
+import "yet_another_pkg"
+option now = () => (2030-01-01T00:00:00Z)
+option foo.bar = "baz"
+builtin foo
+
+test aggregate_window_empty = () => ({
+    input: testing.loadStorage(csv: inData),
+    want: testing.loadMem(csv: outData),
+    fn: (table=<-) =>
+        table
+            |> range(start: 2018-05-22T19:53:26Z, stop: 2018-05-22T19:55:00Z)
+            |> aggregateWindow(every: 30s, fn: sum),
+})
+"#,
+        ),
+        crate::parser::parse_string(
+            "test2",
+            r#"
+a
+
+arr = [0, 1, 2]
+f = (i) => i
+ff = (i=<-, j) => {
+  k = i + j
+  return k
+}
+b = z and y
+b = z or y
+o = {red: "red", "blue": 30}
+m = o.red
+i = arr[0]
+n = 10 - 5 + 10
+n = 10 / 5 * 10
+m = 13 % 3
+p = 2^10
+b = 10 < 30
+b = 10 <= 30
+b = 10 > 30
+b = 10 >= 30
+eq = 10 == 10
+neq = 11 != 10
+b = not false
+e = exists o.red
+tables |> f()
+fncall = id(v: 20)
+fncall2 = foo(v: 20, w: "bar")
+v = if true then 70.0 else 140.0 
+ans = "the answer is ${v}"
+paren = (1)
+
+i = 1
+f = 1.0
+s = "foo"
+d = 10s
+b = true
+dt = 2030-01-01T00:00:00Z
+re =~ /foo/
+re !~ /foo/
+"#,
+        ),
+    ];
+    let pkg = ast::Package {
+        base: ast::BaseNode {
+            ..ast::BaseNode::default()
+        },
+        path: String::from("./"),
+        package: String::from("test"),
+        files: f,
+    };
+    let mut pkg = match analyze(pkg) {
```

Comment:
Does this run type inference, so that the semantic graph you're serializing has types?  I guess that it won't, since there are some undefined symbols there.  Can you file a follow up issue to modify this test to run type inference?

Reply:
Thinking this over, I don't think you need to run type inference in this test.  There are already unit tests for serializing types, so what you have here is fine.  I think we will want to run type inference as part of the roundtrip testing though.